ambry-store/src/main/java/com/github/ambry/store/StoreMessageReadSet.java

```diff
@@ -165,8 +166,11 @@ void doPrefetch(long relativeOffset, long size) throws IOException {
     prefetchedData = PooledByteBufAllocator.DEFAULT.ioBuffer((int) sizeToRead);
     long fetchStartTime = SystemTime.getInstance().milliseconds();
     prefetchedData.writeBytes(getChannel(), offset.getOffset() + relativeOffset, (int) sizeToRead);
-    segment.getMetrics().diskReadTimePerMbInMs.update(
-        ((SystemTime.getInstance().milliseconds() - fetchStartTime) << 20) / sizeToRead);
+    String diskMountPath = segment.getDiskMountPath();
```

Comment:
Similar comment here. The LogSegment could have a member variable with it's metrics and could expose `updateDiskReadTime` and `updateDiskWriteTime` methods instead of looking up in the hashmap every time